src/main/java/com/sandeep/demoemployee/service/EmployeeService.java

```diff
@@ -0,0 +1,225 @@
+package com.sandeep.demoemployee.service;
+
+import com.sandeep.demoemployee.entity.CrudeEmployee;
+import com.sandeep.demoemployee.entity.Designation;
+import com.sandeep.demoemployee.entity.Employee;
+import com.sandeep.demoemployee.repository.DesignationRepository;
+import com.sandeep.demoemployee.repository.EmployeeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class EmployeeService {
+    private final EmployeeRepository employeeRepository;
+    private final DesignationRepository designationRepository;
+
+    @Autowired
+    public EmployeeService(EmployeeRepository employeeRepository, DesignationRepository designationRepository) {
+        this.employeeRepository = employeeRepository;
+        this.designationRepository = designationRepository;
+    }
+
+    // get the list of all employees
+    public List<Employee> getAllEmployees() {
+        return new ArrayList<>(employeeRepository.findAllByOrderByDesignation_lvlIdAscEmpNameAsc());
+    }
+
+    //check is given employee exists or not
+    public boolean employeeExists(int id) {
+        return !employeeRepository.existsAllByEmpIdIs(id);
+    }
+
+    //converting the data to usable format
+    public Employee getEmpFromCrudeEmp(CrudeEmployee crudeEmployee) {
+        Employee employee = new Employee();
+        if (crudeEmployee.getManagerId() != null)
+            employee.setManagerId(crudeEmployee.getManagerId());
+        if (crudeEmployee.getEmpName() != null)
+            employee.setEmpName(crudeEmployee.getEmpName());
+        Designation designation = null;
+        if (crudeEmployee.getDesignation() != null)
+            designation = designationRepository.getByRoleLike(crudeEmployee.getDesignation());
+        employee.setDesignation(designation);
+        return employee;
+    }
+
+    private List<Employee> getAllByManagerId(Integer id) {
+        if (id != 0) {
+            return employeeRepository.findAllByManagerIdOrderByDesignation_lvlIdAscEmpNameAsc(id);
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    //returns total number of employees by designation (was used for debugging)
+    public Long getTotalEmployeeByDesignation(Integer id) {
+        return employeeRepository.countAllByDesignation_DsgnId(id);
+    }
```

Comment:
it is Good that you have provided comment about method used for debugging. 
But it will be a better practice to not checkin the code into repo, if this is for debugging. You should either create git patch or stash and keep it on your local or in a different branch than master.